stacker/util.py

```diff
@@ -448,3 +452,168 @@ def ensure_s3_bucket(s3_client, bucket_name, bucket_region):
+                                        Default to $HOME/.stacker
+        """
+        stacker_cache_dir = (stacker_cache_dir or
+                             os.path.join(os.environ['HOME'], '.stacker'))
+        package_cache_dir = os.path.join(stacker_cache_dir, 'packages')
+        self.stacker_cache_dir = stacker_cache_dir
+        self.package_cache_dir = package_cache_dir
+        self.create_cache_directories()
+
+    def create_cache_directories(self):
+        """Ensures that SourceProcessor cache directories exist.
+
+           Returns True to indicate the cache directories are present."""
+        if not os.path.isdir(self.package_cache_dir):
+            if not os.path.isdir(self.stacker_cache_dir):
+                os.mkdir(self.stacker_cache_dir)
+            os.mkdir(self.package_cache_dir)
+        return True
+
+    def get_package_sources(self, sources):
+        """Makes remote python packages available for local use
+
+        Args:
+            sources (dict): Dictionary of remote sources from config.
+                            Currently supports git repositories
+            Example:
+              {'git': [
+                  {'uri': 'git@github.com:remind101/stacker_blueprints.git',
+                   'tag': '1.0.0',
+                   'paths': ['stacker_blueprints']},
+                  {'uri': 'git@github.com:acmecorp/stacker_blueprints.git'},
+                  {'uri': 'git@github.com:contoso/webapp.git',
+                   'branch': 'staging'},
+                  {'uri': 'git@github.com:contoso/foo.git',
+                   'commit': '12345678'}
+              ]}
+
+        """
+        # Checkout git repositories specified in config
+        if 'git' in sources:
+            for config in sources['git']:
+                self.fetch_git_package(config=config)
+
+    def fetch_git_package(self, config):
+        """Makes a remote git repository available for local use
+
+        Args:
+            config (dict): Dictionary of git repo configuration
+
+        """
+        # If a specific commit or tag is defined, first check to see if it is
+        # already cached.
+        if config.get('commit'):
+            ref = config['commit']
+            dir_name = self.sanitize_git_path(uri=config['uri'],
+                                              ref=ref)
+        elif config.get('tag'):
+            ref = config['tag']
+            dir_name = self.sanitize_git_path(uri=config['uri'],
+                                              ref=ref)
+        else:
+            ref = self.git_ls_remote(
+                config['uri'],
+                self.determine_git_ls_remote_ref(config)
+            )
+            dir_name = self.sanitize_git_path(uri=config['uri'],
+                                              ref=ref)
+
+        cached_dir_path = os.path.join(self.package_cache_dir, dir_name)
+
+        # Clone the repo if it doesn't already exist
+        if not os.path.isdir(cached_dir_path):
```

Comment:
What happens here if I change the tag in the config?  Will the cached dir be updated?  It doesn't seem like it here, but I may be missing something.